src/NServiceBus.Core/Transports/Msmq/Config/MsmqTransportConfigurator.cs

```diff
@@ -163,10 +112,50 @@ protected override void Configure(FeatureConfigurationContext context, string co
             }
 
             var messageLabelGenerator = context.Settings.GetMessageLabelGenerator();
-            context.Container.ConfigureComponent(b=>new MsmqMessageSender(settings, messageLabelGenerator), DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new MsmqMessageSender(settings, messageLabelGenerator), DependencyLifecycle.InstancePerCall);
 
             context.Container.ConfigureComponent<MsmqQueueCreator>(DependencyLifecycle.InstancePerCall)
                 .ConfigureProperty(t => t.Settings, settings);
+
+
+            if (context.Settings.GetOrDefault<bool>("Endpoint.SendOnly"))
+            {
+                return;
+            }
+
+            //todo: pass this in the deque settings instead
+            var transactionSettings = new Unicast.Transport.TransactionSettings(context.Settings);
+
+            context.Container.ConfigureComponent(b =>
+            {
+                MsmqReceiveStrategy receiveStrategy;
```

Comment:
I contemplated that as well, I wonder if we should pass the transaction setting to the IDequeuMessages.Init? This would allow satellites to have different TX levels. Eg. even though the endpoint use DTC the TM satellites can still drow down to native msmq tx and get better perf. cc @SzymonPobiega